DEVELOPMENT.md

```diff
@@ -33,6 +33,24 @@ To run linting, run the following command:
 poetry run ruff .
 ```
 
+### Run tests
+
+We use pytest to test our code, coverage.py to generate coverage reports and [coverage gutters](https://marketplace.visualstudio.com/items?itemName=semasquare.vscode-coverage-gutters#:~:text=Features.%20Supports%20any%20language%20as%20long%20as%20you) VSCode extension to visualize code coverage in the editor.
+
+To run all tests and generate coverage data run the following command:
+
+```bash
+poetry run coverage run --source=src -m pytest .
+```
+
+This will generate a .coverage file. To visualize code coverage, generate a coverage.xml file from the .coverage file by running:
+
+```bash
+poetry run coverage xml
```

Comment:
Awesome, works immediately as expected :) 

Super super super nit: we could chain the commands: `poetry run coverage run --source=src -m pytest . && coverage xml`

Reply:
Done!